sdk/core/Azure.Core/src/RequestFailedException.cs

```diff
@@ -128,6 +134,8 @@ public override void GetObjectData(SerializationInfo info, StreamingContext cont
 
         internal static (string FormattedError, string? ErrorCode, IDictionary<string, string>? Data) GetRequestFailedExceptionContent(Response response)
         {
+            BufferResponseIfNeeded(response);
```

Comment:
Does this need to happen before the exception is thrown, or could it happen lazily if the property is accessed by the caller?

Reply:
I think we need to do it up front as this is where the network stream would be disposed. Otherwise, it won't be disposed unless the user calls the method/property getter.

Reply:
Ah, cool.  Does that mean that we have an issue currently in any libraries that use unbuffered streams and throw an exception without property buffering?